---
title: Pocillopora Samples Environmental analyses - Cladocopium C1 reference
  transcriptome - 16.05.2019
author: "Julie LÃª-Hoang"
output: pdf_document
---

# 1. Load data and packages

## 1.1. Load Packages

```{r}
#install.packages("ggplot2")
library("ggplot2")
#install.packages("stringr", dependencies = TRUE)
library("stringr")
#install.packages("RColorBrewer")
library("RColorBrewer")
#install.packages("dplyr")
library("dplyr")
#install.packages("reshape2")
library("reshape2")
#install.packages("ggrepel")
library("ggrepel")
#install.packages("FactoMineR")
library("FactoMineR")
#install.packages("pheatmap")
library("pheatmap")
library("htmlwidgets")
library("stringi")

## try http:// if https:// URLs are not supported
#source("http://bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library("DESeq2")
#biocLite("edgeR")
library("edgeR")
#biocLite("limma")
library("limma")
#install.packages("gplots")
library("gplots")

#install.packages("venneuler")
library("venneuler")

#install.packages("VennDiagram")
library("VennDiagram")

#install.packages("eulerr")
library("eulerr")
#install.packages("factoextra")
library("factoextra")
#install.packages("treemap")
library("treemap")
#install.packages("GOplot")
library("GOplot")

#install.packages("vegan")
library("vegan")

#install.packages("xlsx") 
#library("xlsx")
#install.packages("seqinr") 
library("seqinr")
#install.packages("openxlsx", dependencies = TRUE)
library(openxlsx)
library(MASS)
```

## 1.2. Load Data

```{r}
#write.table(tabTPM, file = "TPM-Pocillopora-Cladocopium-C1.txt", quote = FALSE, sep = "\t", row.names = TRUE, col.names = TRUE)

#write.table(dataAll, file = "Raw-Count-Pocillopora-Cladocopium-C1.txt", quote = FALSE, sep = "\t", row.names = TRUE, col.names = TRUE)

tabTPM1 <- read.table("TPM-Pocillopora-Cladocopium-C1-CDS.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
tabTPM <- as.matrix(tabTPM1)

dataAll1 <- read.table("Raw-Count-Pocillopora-Cladocopium-C1-CDS.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
dataAll <- as.matrix(dataAll1)
```

```{r}
myDataSites <- read.table("DEG-Comparisons-Between-Sites-Pocillopora-Cladocopium-C1-All-Filtered.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)

# Remove NA in pvalue
filterResSit <- myDataSites[!is.na(myDataSites$padj),]
# Filter p.value
tabSit <- filterResSit[filterResSit$padj < 0.001,]
# Calculs
tabSit$FoldChange <- 2^tabSit$log2FoldChange
tabSit$log10FoldChange <- log10(tabSit$FoldChange)
# order
tabSit <- tabSit[order(tabSit$FoldChange, decreasing = TRUE),]
head(tabSit)

tabSites <- data.frame(tabSit)
rownames(tabSites) <- c(1:nrow(tabSit))
```

##  2.2. Between islands

```{r}
myDataIslands <- read.table("DEG-Comparisons-Between-Islands-Pocillopora-Cladocopium-C1-All-Filtered.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)

# Remove NA in pvalue
filterResIsl <- myDataIslands[!is.na(myDataIslands$padj),]
# Filter p.value
tabIsl <- filterResIsl[filterResIsl$padj < 0.001,]
# Calculs
tabIsl$FoldChange <- 2^tabIsl$log2FoldChange
tabIsl$log10FoldChange <- log10(tabIsl$FoldChange)
# order
tabIsl <- tabIsl[order(tabIsl$FoldChange, decreasing = TRUE),]
head(tabIsl)

tabIslands <- data.frame(tabIsl)
rownames(tabIslands) <- c(1:nrow(tabIsl))
```

# 3. Differentially expressed transcripts

## 3.1. Data By Sites

```{r}
tabSites$log2FoldChange <- as.numeric(tabSites$log2FoldChange)
myData.fil <- tabSites[tabSites$log2FoldChange >= 2 | tabSites$log2FoldChange < -2,]

# Key Numbers
length(unique(tabSites[order(tabSites$Genes, decreasing =  FALSE),1]))
length(unique(myData.fil[order(myData.fil$Genes, decreasing =  FALSE),1]))

# Colors
my_col <- c(rep("#9E0142",9), rep("#F46D43",9), rep("#FEE08B",10), rep("#ABDDA4",12), rep("#3288BD",9), rep("#5E4FA2",9), rep("#C51B7D", 9), rep("#4D9221",9), rep("#8C510A",9), rep("#CAB2D6",9), rep("#5E4FA2",9))
col<- colorRampPalette(c("red", "white", "blue"))(256)
colors <- c("#9E0142", "#F46D43", "#FEE08B", "#ABDDA4", "#3288BD", "#5E4FA2", "#C51B7D", "#4D9221", "#8C510A", "#CAB2D6", "#66C2A5")#, "#5E4FA2")

# Fold Change filter
# Heatmap
listDEG <- as.vector(unique(myData.fil[order(myData.fil$Genes, decreasing =  FALSE),1]))
# TPM
DEG.TPM <- tabTPM[rownames(tabTPM)%in%listDEG,]
DEG.TPM.matrix <- as.matrix(tabTPM[rownames(tabTPM)%in%listDEG,])
test <- DEG.TPM.matrix
for(j in 1:ncol(test)){
  for(i in 1:nrow(test)){
    if(test[i,j] == 0){
      #print(test[i,j])
      test[i,j] <- 0.01
    }
  }
}
DEG.TPM.melt <- melt(test)
namesGenesOrder.TPM <- rownames(DEG.TPM[hclust(dist(DEG.TPM))$order,])
otherOrderTabmyData.TPM <- melt(DEG.TPM)
otherOrderTabmyData.TPM$Var1 <- factor(otherOrderTabmyData.TPM$Var1, levels = namesGenesOrder.TPM)
conditionIsl <- rep(names(table(substr(colnames(DEG.TPM.matrix), 1, 3))), table(substr(colnames(DEG.TPM.matrix), 1, 3)))

# Raw Count
DEG.Raw <- dataAll[rownames(dataAll)%in%listDEG,]
DEG.Raw.matrix <- as.matrix(dataAll[rownames(dataAll)%in%listDEG,])
test <- DEG.Raw.matrix
for(j in 1:ncol(test)){
  for(i in 1:nrow(test)){
    if(test[i,j] == 0){
      #print(test[i,j])
      test[i,j] <- 0.01
    }
  }
}
DEG.Raw.melt <- melt(test)
namesGenesOrder.Raw <- rownames(DEG.Raw[hclust(dist(DEG.Raw))$order,])
otherOrderTabmyData.Raw <- melt(DEG.Raw)
otherOrderTabmyData.Raw$Var1 <- factor(otherOrderTabmyData.Raw$Var1, levels = namesGenesOrder.Raw)

dis <- vegdist(t(DEG.TPM))

## Fold Change without filter
# Heatmap
listDEG <- as.vector(unique(tabSites[order(tabSites$Genes, decreasing =  FALSE),1]))
# TPM2
DEG.TPM2 <- tabTPM[rownames(tabTPM)%in%listDEG,]
DEG.TPM2.matrix <- as.matrix(tabTPM[rownames(tabTPM)%in%listDEG,])
test <- DEG.TPM2.matrix
for(j in 1:ncol(test)){
  for(i in 1:nrow(test)){
    if(test[i,j] == 0){
      #print(test[i,j])
      test[i,j] <- 0.01
    }
  }
}
DEG.TPM2.melt <- melt(test)
namesGenesOrder.TPM2 <- rownames(DEG.TPM2[hclust(dist(DEG.TPM2))$order,])
otherOrderTabmyData.TPM2 <- melt(DEG.TPM2)
otherOrderTabmyData.TPM2$Var1 <- factor(otherOrderTabmyData.TPM2$Var1, levels = namesGenesOrder.TPM2)
conditionIsl <- rep(names(table(substr(colnames(DEG.TPM2.matrix), 1, 3))), table(substr(colnames(DEG.TPM2.matrix), 1, 3)))

# Raw2 Count
DEG.Raw2 <- dataAll[rownames(dataAll)%in%listDEG,]
DEG.Raw2.matrix <- as.matrix(dataAll[rownames(dataAll)%in%listDEG,])
test <- DEG.Raw2.matrix
for(j in 1:ncol(test)){
  for(i in 1:nrow(test)){
    if(test[i,j] == 0){
      #print(test[i,j])
      test[i,j] <- 0.01
    }
  }
}
DEG.Raw2.melt <- melt(test)
namesGenesOrder.Raw2 <- rownames(DEG.Raw2[hclust(dist(DEG.Raw2))$order,])
otherOrderTabmyData.Raw2 <- melt(DEG.Raw2)
otherOrderTabmyData.Raw2$Var1 <- factor(otherOrderTabmyData.Raw2$Var1, levels = namesGenesOrder.Raw2)

dis2 <- vegdist(t(DEG.TPM2))
```

## 3.2. Data By Islands

```{r}
tabIslands$log2FoldChange <- as.numeric(tabIslands$log2FoldChange)
myData.fil2 <- tabIslands[tabIslands$log2FoldChange >= 2 | tabIslands$log2FoldChange < -2,]

# Key Numbers
length(unique(tabIslands[order(tabIslands$Genes, decreasing =  FALSE),1]))
length(unique(myData.fil2[order(myData.fil2$Genes, decreasing =  FALSE),1]))

# Fold Change filter
# Heatmap
listDEG2 <- as.vector(unique(myData.fil2[order(myData.fil2$Genes, decreasing =  FALSE),1]))
# TPM.Isl
DEG.TPM.Isl <- tabTPM[rownames(tabTPM)%in%listDEG2,]
DEG.TPM.Isl.matrix <- as.matrix(tabTPM[rownames(tabTPM)%in%listDEG2,])
test <- DEG.TPM.Isl.matrix
for(j in 1:ncol(test)){
  for(i in 1:nrow(test)){
    if(test[i,j] == 0){
      #print(test[i,j])
      test[i,j] <- 0.01
    }
  }
}
DEG.TPM.Isl.melt <- melt(test)
namesGenesOrder.TPM.Isl <- rownames(DEG.TPM.Isl[hclust(dist(DEG.TPM.Isl))$order,])
otherOrderTabmyData.TPM.Isl <- melt(DEG.TPM.Isl)
otherOrderTabmyData.TPM.Isl$Var1 <- factor(otherOrderTabmyData.TPM.Isl$Var1, levels = namesGenesOrder.TPM.Isl)
conditionIsl <- rep(names(table(substr(colnames(DEG.TPM.Isl.matrix), 1, 3))), table(substr(colnames(DEG.TPM.Isl.matrix), 1, 3)))

# Raw.Isl Count
DEG.Raw.Isl <- dataAll[rownames(dataAll)%in%listDEG2,]
DEG.Raw.Isl.matrix <- as.matrix(dataAll[rownames(dataAll)%in%listDEG2,])
test <- DEG.Raw.Isl.matrix
for(j in 1:ncol(test)){
  for(i in 1:nrow(test)){
    if(test[i,j] == 0){
      #print(test[i,j])
      test[i,j] <- 0.01
    }
  }
}
DEG.Raw.Isl.melt <- melt(test)
namesGenesOrder.Raw.Isl <- rownames(DEG.Raw.Isl[hclust(dist(DEG.Raw.Isl))$order,])
otherOrderTabmyData.Raw.Isl <- melt(DEG.Raw.Isl)
otherOrderTabmyData.Raw.Isl$Var1 <- factor(otherOrderTabmyData.Raw.Isl$Var1, levels = namesGenesOrder.Raw.Isl)

disIsl <- vegdist(t(DEG.TPM.Isl))

## Fold Change without filter
# Heatmap
listDEG2 <- as.vector(unique(tabIslands[order(tabIslands$Genes, decreasing =  FALSE),1]))
# TPM.Isl2
DEG.TPM.Isl2 <- tabTPM[rownames(tabTPM)%in%listDEG2,]
DEG.TPM.Isl2.matrix <- as.matrix(tabTPM[rownames(tabTPM)%in%listDEG2,])
test <- DEG.TPM.Isl2.matrix
for(j in 1:ncol(test)){
  for(i in 1:nrow(test)){
    if(test[i,j] == 0){
      #print(test[i,j])
      test[i,j] <- 0.01
    }
  }
}
DEG.TPM.Isl2.melt <- melt(test)
namesGenesOrder.TPM.Isl2 <- rownames(DEG.TPM.Isl2[hclust(dist(DEG.TPM.Isl2))$order,])
otherOrderTabmyData.TPM.Isl2 <- melt(DEG.TPM.Isl2)
otherOrderTabmyData.TPM.Isl2$Var1 <- factor(otherOrderTabmyData.TPM.Isl2$Var1, levels = namesGenesOrder.TPM.Isl2)
conditionIsl <- rep(names(table(substr(colnames(DEG.TPM.Isl2.matrix), 1, 3))), table(substr(colnames(DEG.TPM.Isl2.matrix), 1, 3)))

# Raw.Isl2 Count
DEG.Raw.Isl2 <- dataAll[rownames(dataAll)%in%listDEG2,]
DEG.Raw.Isl2.matrix <- as.matrix(dataAll[rownames(dataAll)%in%listDEG2,])
test <- DEG.Raw.Isl2.matrix
for(j in 1:ncol(test)){
  for(i in 1:nrow(test)){
    if(test[i,j] == 0){
      #print(test[i,j])
      test[i,j] <- 0.01
    }
  }
}
DEG.Raw.Isl2.melt <- melt(test)
namesGenesOrder.Raw.Isl2 <- rownames(DEG.Raw.Isl2[hclust(dist(DEG.Raw.Isl2))$order,])
otherOrderTabmyData.Raw.Isl2 <- melt(DEG.Raw.Isl2)
otherOrderTabmyData.Raw.Isl2$Var1 <- factor(otherOrderTabmyData.Raw.Isl2$Var1, levels = namesGenesOrder.Raw.Isl2)

disIsl2 <- vegdist(t(DEG.TPM.Isl2))
```

# 4. Environmental parameters

```{r}
# Load environmental data
res <- read.xlsx("EnvData.xlsx", sheet = 4, colNames=TRUE)
env1 <- data.frame(as.matrix(res, nrow = nrow(res)), stringsAsFactors = FALSE)
env2 <- data.frame(apply(env1[3:ncol(env1)], 2, as.numeric), stringsAsFactors = FALSE)
rownames(env2) <- colnames(tabTPM[,c(grep("I04",colnames(tabTPM)), grep("I05",colnames(tabTPM)), grep("I06",colnames(tabTPM)), grep("I08",colnames(tabTPM)), grep("I09",colnames(tabTPM)), grep("I10",colnames(tabTPM)))])
env2$Ile <- substr(rownames(env2), 1, 3)
env2$Samples <- rownames(env2)
env <- env2

colo <- c("#67001F","#053061", "#D6604D","#4393C3","#C51B7D","#4D9221","#762A83","#BF812D")

pdf(file = "Environmental-parameters-Cladocopium-C1-Pocillopora.pdf", width = 15)
## Primary physicochemical variables
ggplot(env, aes(x = Samples))+geom_point(aes(y = Temp, col = "Temp"))+geom_point(aes(y = Chla, col = "Chla"))+geom_point(aes(y = PAR, col = "PAR"))+geom_point(aes(y = Primary_Production_g_par_m3, col = "Primary_Production_g_par_m3"))+geom_point(aes(y = O2, col = "O2"))+geom_point(aes(y = Salinity, col = "Salinity"))+geom_point(aes(y = PIC, col = "PIC"))+geom_point(aes(y = POC, col = "POC"))+facet_grid(.~Ile, scales = "free_x")+labs(title = "Primary physicochemical variables", x = "Samples", y = "Data")+scale_color_manual(breaks = c("Chla", "O2", "PAR","PIC","POC","Primary_Production_g_par_m3","Salinity","Temp"), values=colo)+theme(axis.text.x = element_text(angle = 90))#+geom_text(data = env, aes(x = env$Samples, y = (env$Chla+2)), label = round(env$Chla, digits = 2), col = colo[1])+geom_text(data = env, aes(x = env$Samples, y = (env$O2-5)), label = round(env$O2, digits = 2), col = colo[2])+geom_text(data = env, aes(x = env$Samples, y = (env$PAR-3)), label = round(env$PAR, digits = 2), col = colo[3])+geom_text(data = env, aes(x = env$Samples, y = (env$PIC-3)), label = round(env$PIC, digits = 2), col = colo[4])+geom_text(data = env, aes(x = env$Samples, y = (env$POC-3)), label = round(env$POC, digits = 2), col = colo[5])+geom_text(data = env, aes(x = env$Samples, y = (env$Primary_Production_g_par_m3-3)), label = round(env$Primary_Production_g_par_m3, digits = 5), col = colo[6])+geom_text(data = env, aes(x = env$Samples, y = (env$Salinity-3)), label = round(env$Salinity, digits = 2), col = colo[7])+geom_text(data = env, aes(x = env$Samples, y = (env$Temp-3)), label = round(env$Temp, digits = 2), col = colo[8])

## Micronutrients
ggplot(env, aes(x = Samples))+geom_point(aes(y = log_Fe_nM, col = "log_Fe_nM"))+geom_point(aes(y = Nitrites_uM, col = "Nitrites_uM"))+geom_point(aes(y = Nitrates_uM, col = "Nitrates_uM"))+geom_point(aes(y = Phosphates_uM, col = "Phosphates_uM"))+geom_point(aes(y = Silice_uM, col = "Silice_uM"))+facet_grid(.~Ile, scales = "free_x")+labs(title = "Micronutrients variables", x = "Samples", y = "Data")+scale_color_manual(breaks = c("log_Fe_nM", "Nitrates_uM", "Nitrites_uM","Phosphates_uM","Silice_uM"), values=colo)+theme(axis.text.x = element_text(angle = 90))#+geom_text(data = env, aes(x = env$Samples, y = (env$log_Fe_nM-1)), label = round(env$log_Fe_nM, digits = 2), col = colo[1])+geom_text(data = env, aes(x = env$Samples, y = (env$Nitrates_uM-0.5)), label = round(env$Nitrates_uM, digits = 2), col = colo[2])+geom_text(data = env, aes(x = env$Samples, y = (env$Nitrites_uM+0.5)), label = round(env$Nitrites_uM, digits = 2), col = colo[3])+geom_text(data = env, aes(x = env$Samples, y = (env$Phosphates_uM+1)), label = round(env$Phosphates_uM, digits = 2), col = colo[4])+geom_text(data = env, aes(x = env$Samples, y = (env$Silice_uM+1)), label = round(env$Silice_uM, digits = 2), col = colo[5])

## Trace Minerals
ggplot(env, aes(x = Samples))+geom_point(aes(y = Cadmium_pM, col = "Cadmium_pM"))+geom_point(aes(y = Cuivre_nM, col = "Cuivre_nM"))+geom_point(aes(y = Cobalt_pM, col = "Cobalt_pM"))+geom_point(aes(y = Magnesium_pM, col = "Magnesium_pM"))+geom_point(aes(y = Nickel_nM, col = "Nickel_nM"))+geom_point(aes(y = Plomb_pM, col = "Plomb_pM"))+geom_point(aes(y = Zinc_nM, col = "Zinc_nM"))+facet_grid(.~Ile, scales = "free_x")+labs(title = "Trace Minerals", x = "Samples", y = "Data")+scale_color_manual(breaks = c("Cadmium_pM", "Cobalt_pM", "Cuivre_nM","Magnesium_pM","Nickel_nM","Plomb_pM","Zinc_nM"), values=colo)+theme(axis.text.x = element_text(angle = 90))#+geom_text(data = env, aes(x = env$Samples, y = (env$Cadmium_pM-1)), label = round(env$Cadmium_pM, digits = 2), col = colo[1])+geom_text(data = env, aes(x = env$Samples, y = (env$Cobalt_pM-0.5)), label = round(env$Cobalt_pM, digits = 2), col = colo[2])#+geom_text(data = env, aes(x = env$Samples, y = (env$Cuivre_nM+0.5)), label = round(env$Cuivre_nM, digits = 2), col = colo[3])+geom_text(data = env, aes(x = env$Samples, y = (env$Magnesium_pM+1)), label = round(env$Magnesium_pM, digits = 2), col = colo[4])+geom_text(data = env, aes(x = env$Samples, y = (env$Nickel_nM+1)), label = round(env$Nickel_nM, digits = 2), col = colo[5])+geom_text(data = env, aes(x = env$Samples, y = (env$Plomb_pM+1)), label = round(env$Plomb_pM, digits = 2), col = colo[6])+geom_text(data = env, aes(x = env$Samples, y = (env$Zinc_nM+1)), label = round(env$Zinc_nM, digits = 2), col = colo[7])
dev.off()
```

## 4.1. By Sites

```{r}
env <- env[,1:21]
# Without filtering FC
# Transform DEG data
data.1 <- t(as.data.frame(data.frame(matrix(DEG.TPM2[,c(grep("I04", colnames(DEG.TPM2)), grep("I05", colnames(DEG.TPM2)), grep("I06", colnames(DEG.TPM2)), grep("I08", colnames(DEG.TPM2)), grep("I09", colnames(DEG.TPM2)), grep("I10", colnames(DEG.TPM2)))], nrow =  nrow(DEG.TPM2[,c(grep("I04", colnames(DEG.TPM2)), grep("I05", colnames(DEG.TPM2)), grep("I06", colnames(DEG.TPM2)), grep("I08", colnames(DEG.TPM2)), grep("I09", colnames(DEG.TPM2)), grep("I10", colnames(DEG.TPM2)))])))))

rownames(data.1) <- colnames(DEG.TPM2[,c(grep("I04", colnames(DEG.TPM2)), grep("I05", colnames(DEG.TPM2)), grep("I06", colnames(DEG.TPM2)), grep("I08", colnames(DEG.TPM2)), grep("I09", colnames(DEG.TPM2)), grep("I10", colnames(DEG.TPM2)))])
colnames(data.1) <- rownames(DEG.TPM2[,c(grep("I04", colnames(DEG.TPM2)), grep("I05", colnames(DEG.TPM2)), grep("I06", colnames(DEG.TPM2)), grep("I08", colnames(DEG.TPM2)), grep("I09", colnames(DEG.TPM2)), grep("I10", colnames(DEG.TPM2)))])
# Keep only DEG which are expressed (due to filtering of islands without environmental parameters)
data.t <- data.1[rowSums(data.1) > 0,]
data <- data.t

# With filtering FC
# Transform DEG data
data2.1 <- t(as.data.frame(data.frame(matrix(DEG.TPM[,c(grep("I04", colnames(DEG.TPM)), grep("I05", colnames(DEG.TPM)), grep("I06", colnames(DEG.TPM)), grep("I08", colnames(DEG.TPM)), grep("I09", colnames(DEG.TPM)), grep("I10", colnames(DEG.TPM)))], nrow =  nrow(DEG.TPM[,c(grep("I04", colnames(DEG.TPM)), grep("I05", colnames(DEG.TPM)), grep("I06", colnames(DEG.TPM)), grep("I08", colnames(DEG.TPM)), grep("I09", colnames(DEG.TPM)), grep("I10", colnames(DEG.TPM)))])))))

rownames(data2.1) <- colnames(DEG.TPM[,c(grep("I04", colnames(DEG.TPM)), grep("I05", colnames(DEG.TPM)), grep("I06", colnames(DEG.TPM)), grep("I08", colnames(DEG.TPM)), grep("I09", colnames(DEG.TPM)), grep("I10", colnames(DEG.TPM)))])
colnames(data2.1) <- rownames(DEG.TPM[,c(grep("I04", colnames(DEG.TPM)), grep("I05", colnames(DEG.TPM)), grep("I06", colnames(DEG.TPM)), grep("I08", colnames(DEG.TPM)), grep("I09", colnames(DEG.TPM)), grep("I10", colnames(DEG.TPM)))])
# Keep only DEG which are expressed (due to filtering of islands without environmental parameters)
data.t2 <- data2.1[rowSums(data2.1) > 0,]
data.2 <- data.t2

# Without filtering FC
# CCA
## All Parameters
cca.1 <- cca(data ~., data = env)
scrs1 <- scores(cca.1, display=c("sp","wa","lc","bp","cn"))

df_Sites1 <- data.frame(scrs1$sites)
colnames(df_Sites1) <- c("x","y")
df_Sites1$Samples <- rownames(df_Sites1)
df_Sites1$Ile <- substr(rownames(data), 1, 3)

multiplier1 <- vegan:::ordiArrowMul(scrs1$biplot)
df_arrows1 <- as.data.frame(scrs1$biplot*multiplier1)
colnames(df_arrows1)<-c("x","y")

## Primary physicochemical variables
cca.2 <- cca(data ~ Temp+Chla+PAR+Primary_Production_g_par_m3+O2+Salinity+PIC+POC, data = env)
scrs2 <- scores(cca.2, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs2)

df_Sites2 <- data.frame(scrs2$sites)
colnames(df_Sites2) <- c("x","y")
df_Sites2$Samples <- rownames(df_Sites2)
df_Sites2$Ile <- substr(rownames(data), 1, 3)

multiplier2 <- vegan:::ordiArrowMul(scrs2$biplot)
df_arrows2 <- as.data.frame(scrs2$biplot*multiplier2)
colnames(df_arrows2)<-c("x","y")

## Micronutrients
cca.3 <- cca(data ~ Nitrates_uM+Nitrites_uM+Phosphates_uM+Silice_uM+log_Fe_nM, data = env)
scrs3 <- scores(cca.3, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs3)

df_Sites3 <- data.frame(scrs3$sites)
colnames(df_Sites3) <- c("x","y")
df_Sites3$Samples <- rownames(df_Sites3)
df_Sites3$Ile <- substr(rownames(data), 1, 3)

multiplier3 <- vegan:::ordiArrowMul(scrs3$biplot)
df_arrows3 <- as.data.frame(scrs3$biplot*multiplier3)
colnames(df_arrows3)<-c("x","y")

## Trace Minerals
cca.4 <- cca(data ~ Zinc_nM+Cadmium_pM+Nickel_nM+Cuivre_nM+Plomb_pM+Magnesium_pM+Cobalt_pM, data = env)
scrs4 <- scores(cca.4, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs4)

df_Sites4 <- data.frame(scrs4$sites)
colnames(df_Sites4) <- c("x","y")
df_Sites4$Samples <- rownames(df_Sites4)
df_Sites4$Ile <- substr(rownames(data), 1, 3)

multiplier4 <- vegan:::ordiArrowMul(scrs4$biplot)
df_arrows4 <- as.data.frame(scrs4$biplot*multiplier4)
colnames(df_arrows4)<-c("x","y")

# Graphs
pdf(file = "CCA-Environmental-parameters-Cladocopium-C1-Pocillopora-Graphs-By-Sites.pdf", width = 12, heigh = 10)
plot(ggplot(df_Sites1, aes(x, y))+geom_point(aes(colour = df_Sites1$Ile))+geom_text(data=df_Sites1, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df_arrows1, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows1*1.1), aes(x, y, label = rownames(df_arrows1)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - All Environmental parameters", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites1, aes(x, y))+geom_point(aes(colour = df_Sites1$Ile))+geom_segment(data = df_arrows1, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows1*1.1), aes(x, y, label = rownames(df_arrows1)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - All Environmental parameters", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites2, aes(x, y))+geom_point(aes(colour = df_Sites2$Ile))+geom_text(data=df_Sites2, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df_arrows2, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows2*1.1), aes(x, y, label = rownames(df_arrows2)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Primary physicochemical variables", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites2, aes(x, y))+geom_point(aes(colour = df_Sites2$Ile))+geom_segment(data = df_arrows2, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows2*1.1), aes(x, y, label = rownames(df_arrows2)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Primary physicochemical variables", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites3, aes(x, y))+geom_point(aes(colour = df_Sites3$Ile))+geom_text(data=df_Sites3, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df_arrows3, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows3*1.1), aes(x, y, label = rownames(df_arrows3)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Micronutrients", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites3, aes(x, y))+geom_point(aes(colour = df_Sites3$Ile))+geom_segment(data = df_arrows3, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows3*1.1), aes(x, y, label = rownames(df_arrows3)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Micronutrients", x= "CCA1", y ="CCA2", fill = "legend"))



plot(ggplot(df_Sites4, aes(x, y))+geom_point(aes(colour = df_Sites4$Ile))+geom_text(data=df_Sites4, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df_arrows4, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows4*1.1), aes(x, y, label = rownames(df_arrows4)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Trace Minerals", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites4, aes(x, y))+geom_point(aes(colour = df_Sites4$Ile))+geom_segment(data = df_arrows4, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows4*1.1), aes(x, y, label = rownames(df_arrows4)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Trace Minerals", x= "CCA1", y ="CCA2", fill = "legend"))
dev.off()

# With filtering FC
# CCA
## All Parameters
cca2.1 <- cca(data.2 ~., data = env)
scrs2.1 <- scores(cca2.1, display=c("sp","wa","lc","bp","cn"))
#scrs2.1 <- scores(cca2.1, display=c("sp","bp"))
#attributes(scrs2.1)

df2_Sites1 <- data.frame(scrs2.1$sites)
colnames(df2_Sites1) <- c("x","y")
df2_Sites1$Samples <- rownames(df2_Sites1)
df2_Sites1$Ile <- substr(rownames(data.2), 1, 3)

multiplier2.1 <- vegan:::ordiArrowMul(scrs2.1$biplot)
df2_arrows1 <- as.data.frame(scrs2.1$biplot*multiplier2.1)
colnames(df2_arrows1)<-c("x","y")

## Primary physicochemical variables
cca2.2 <- cca(data.2 ~ Temp+Chla+PAR+Primary_Production_g_par_m3+O2+Salinity+PIC+POC, data = env)
scrs2.2 <- scores(cca2.2, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs2.2)

df2_Sites2 <- data.frame(scrs2.2$sites)
colnames(df2_Sites2) <- c("x","y")
df2_Sites2$Samples <- rownames(df2_Sites2)
df2_Sites2$Ile <- substr(rownames(data.2), 1, 3)

multiplier2.2 <- vegan:::ordiArrowMul(scrs2.2$biplot)
df2_arrows2 <- as.data.frame(scrs2.2$biplot*multiplier2.2)
colnames(df2_arrows2)<-c("x","y")

## Micronutrients
cca2.3 <- cca(data.2 ~ Nitrates_uM+Nitrites_uM+Phosphates_uM+Silice_uM+log_Fe_nM, data = env)
scrs2.3 <- scores(cca2.3, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs2.3)

df2_Sites3 <- data.frame(scrs2.3$sites)
colnames(df2_Sites3) <- c("x","y")
df2_Sites3$Samples <- rownames(df2_Sites3)
df2_Sites3$Ile <- substr(rownames(data.2), 1, 3)

multiplier2.3 <- vegan:::ordiArrowMul(scrs2.3$biplot)
df2_arrows3 <- as.data.frame(scrs2.3$biplot*multiplier2.3)
colnames(df2_arrows3)<-c("x","y")

## Trace Minerals
cca2.4 <- cca(data.2 ~ Zinc_nM+Cadmium_pM+Nickel_nM+Cuivre_nM+Plomb_pM+Magnesium_pM+Cobalt_pM, data = env)
scrs2.4 <- scores(cca2.4, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs2.4)

df2_Sites4 <- data.frame(scrs2.4$sites)
colnames(df2_Sites4) <- c("x","y")
df2_Sites4$Samples <- rownames(df2_Sites4)
df2_Sites4$Ile <- substr(rownames(data.2), 1, 3)

multiplier2.4 <- vegan:::ordiArrowMul(scrs2.4$biplot)
df2_arrows4 <- as.data.frame(scrs2.4$biplot*multiplier2.4)
colnames(df2_arrows4)<-c("x","y")

# Graphs
pdf(file = "CCA-Environmental-parameters-Cladocopium-C1-Pocillopora-Graphs-With-Filter-By-Sites.pdf", width = 14, heigh = 12)
plot(ggplot(df2_Sites1, aes(x, y))+geom_point(aes(colour = df2_Sites1$Ile))+geom_text(data=df2_Sites1, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df2_arrows1, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows1*1.1), aes(x, y, label = rownames(df2_arrows1)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - All Environmental parameters - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites1, aes(x, y))+geom_point(aes(colour = df2_Sites1$Ile))+geom_segment(data = df2_arrows1, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows1*1.1), aes(x, y, label = rownames(df2_arrows1)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - All Environmental parameters - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites2, aes(x, y))+geom_point(aes(colour = df2_Sites2$Ile))+geom_text(data=df2_Sites2, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df2_arrows2, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows2*1.1), aes(x, y, label = rownames(df2_arrows2)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Primary physicochemical variables - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites2, aes(x, y))+geom_point(aes(colour = df2_Sites2$Ile))+geom_segment(data = df2_arrows2, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows2*1.1), aes(x, y, label = rownames(df2_arrows2)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Primary physicochemical variables - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites3, aes(x, y))+geom_point(aes(colour = df2_Sites3$Ile))+geom_text(data=df2_Sites3, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df2_arrows3, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows3*1.1), aes(x, y, label = rownames(df2_arrows3)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Micronutrients - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites3, aes(x, y))+geom_point(aes(colour = df2_Sites3$Ile))+geom_segment(data = df2_arrows3, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows3*1.1), aes(x, y, label = rownames(df2_arrows3)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Micronutrients - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites4, aes(x, y))+geom_point(aes(colour = df2_Sites4$Ile))+geom_text(data=df2_Sites4, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df2_arrows4, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows4*1.1), aes(x, y, label = rownames(df2_arrows4)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Trace Minerals - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites4, aes(x, y))+geom_point(aes(colour = df2_Sites4$Ile))+geom_segment(data = df2_arrows4, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows4*1.1), aes(x, y, label = rownames(df2_arrows4)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Trace Minerals - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))
dev.off()
```

## 4.2. By Islands

```{r}
env <- env[,1:20]
# Without filtering FC
# Transform DEG data
data.1 <- t(as.data.frame(data.frame(matrix(DEG.TPM.Isl2[,c(grep("I04", colnames(DEG.TPM.Isl2)), grep("I05", colnames(DEG.TPM.Isl2)), grep("I06", colnames(DEG.TPM.Isl2)), grep("I08", colnames(DEG.TPM.Isl2)), grep("I09", colnames(DEG.TPM.Isl2)), grep("I10", colnames(DEG.TPM.Isl2)))], nrow =  nrow(DEG.TPM.Isl2[,c(grep("I04", colnames(DEG.TPM.Isl2)), grep("I05", colnames(DEG.TPM.Isl2)), grep("I06", colnames(DEG.TPM.Isl2)), grep("I08", colnames(DEG.TPM.Isl2)), grep("I09", colnames(DEG.TPM.Isl2)), grep("I10", colnames(DEG.TPM.Isl2)))])))))

rownames(data.1) <- colnames(DEG.TPM.Isl2[,c(grep("I04", colnames(DEG.TPM.Isl2)), grep("I05", colnames(DEG.TPM.Isl2)), grep("I06", colnames(DEG.TPM.Isl2)), grep("I08", colnames(DEG.TPM.Isl2)), grep("I09", colnames(DEG.TPM.Isl2)), grep("I10", colnames(DEG.TPM.Isl2)))])
colnames(data.1) <- rownames(DEG.TPM.Isl2[,c(grep("I04", colnames(DEG.TPM.Isl2)), grep("I05", colnames(DEG.TPM.Isl2)), grep("I06", colnames(DEG.TPM.Isl2)), grep("I08", colnames(DEG.TPM.Isl2)), grep("I09", colnames(DEG.TPM.Isl2)), grep("I10", colnames(DEG.TPM.Isl2)))])
# Keep only DEG which are expressed (due to filtering of islands without environmental parameters)
data.t <- data.1[rowSums(data.1) > 0,]
data <- data.t

# With filtering FC
# Transform DEG data
data2.1 <- t(as.data.frame(data.frame(matrix(DEG.TPM.Isl[,c(grep("I04", colnames(DEG.TPM.Isl)), grep("I05", colnames(DEG.TPM.Isl)), grep("I06", colnames(DEG.TPM.Isl)), grep("I08", colnames(DEG.TPM.Isl)), grep("I09", colnames(DEG.TPM.Isl)), grep("I10", colnames(DEG.TPM.Isl)))], nrow =  nrow(DEG.TPM.Isl[,c(grep("I04", colnames(DEG.TPM.Isl)), grep("I05", colnames(DEG.TPM.Isl)), grep("I06", colnames(DEG.TPM.Isl)), grep("I08", colnames(DEG.TPM.Isl)), grep("I09", colnames(DEG.TPM.Isl)), grep("I10", colnames(DEG.TPM.Isl)))])))))

rownames(data2.1) <- colnames(DEG.TPM.Isl[,c(grep("I04", colnames(DEG.TPM.Isl)), grep("I05", colnames(DEG.TPM.Isl)), grep("I06", colnames(DEG.TPM.Isl)), grep("I08", colnames(DEG.TPM.Isl)), grep("I09", colnames(DEG.TPM.Isl)), grep("I10", colnames(DEG.TPM.Isl)))])
colnames(data2.1) <- rownames(DEG.TPM.Isl[,c(grep("I04", colnames(DEG.TPM.Isl)), grep("I05", colnames(DEG.TPM.Isl)), grep("I06", colnames(DEG.TPM.Isl)), grep("I08", colnames(DEG.TPM.Isl)), grep("I09", colnames(DEG.TPM.Isl)), grep("I10", colnames(DEG.TPM.Isl)))])
# Keep only DEG which are expressed (due to filtering of islands without environmental parameters)
data.t2 <- data2.1[rowSums(data2.1) > 0,]
data.2 <- data.t2

# Without filtering FC
# CCA
## All Parameters
cca.1 <- cca(data ~., data = env)
scrs1 <- scores(cca.1, display=c("sp","wa","lc","bp","cn"))

df_Sites1 <- data.frame(scrs1$sites)
colnames(df_Sites1) <- c("x","y")
df_Sites1$Samples <- rownames(df_Sites1)
df_Sites1$Ile <- substr(rownames(data), 1, 3)

multiplier1 <- vegan:::ordiArrowMul(scrs1$biplot)
df_arrows1 <- as.data.frame(scrs1$biplot*multiplier1)
colnames(df_arrows1)<-c("x","y")

## Primary physicochemical variables
cca.2 <- cca(data ~ Temp+Chla+PAR+Primary_Production_g_par_m3+O2+Salinity+PIC+POC, data = env)
scrs2 <- scores(cca.2, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs2)

df_Sites2 <- data.frame(scrs2$sites)
colnames(df_Sites2) <- c("x","y")
df_Sites2$Samples <- rownames(df_Sites2)
df_Sites2$Ile <- substr(rownames(data), 1, 3)

multiplier2 <- vegan:::ordiArrowMul(scrs2$biplot)
df_arrows2 <- as.data.frame(scrs2$biplot*multiplier2)
colnames(df_arrows2)<-c("x","y")

## Micronutrients
cca.3 <- cca(data ~ Nitrates_uM+Nitrites_uM+Phosphates_uM+Silice_uM+log_Fe_nM, data = env)
scrs3 <- scores(cca.3, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs3)

df_Sites3 <- data.frame(scrs3$sites)
colnames(df_Sites3) <- c("x","y")
df_Sites3$Samples <- rownames(df_Sites3)
df_Sites3$Ile <- substr(rownames(data), 1, 3)

multiplier3 <- vegan:::ordiArrowMul(scrs3$biplot)
df_arrows3 <- as.data.frame(scrs3$biplot*multiplier3)
colnames(df_arrows3)<-c("x","y")

## Trace Minerals
cca.4 <- cca(data ~ Zinc_nM+Cadmium_pM+Nickel_nM+Cuivre_nM+Plomb_pM+Magnesium_pM+Cobalt_pM, data = env)
scrs4 <- scores(cca.4, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs4)

df_Sites4 <- data.frame(scrs4$sites)
colnames(df_Sites4) <- c("x","y")
df_Sites4$Samples <- rownames(df_Sites4)
df_Sites4$Ile <- substr(rownames(data), 1, 3)

multiplier4 <- vegan:::ordiArrowMul(scrs4$biplot)
df_arrows4 <- as.data.frame(scrs4$biplot*multiplier4)
colnames(df_arrows4)<-c("x","y")

# Graphs
pdf(file = "CCA-Environmental-parameters-Cladocopium-C1-Pocillopora-Graphs-By-Islands.pdf", width = 12, heigh = 10)
plot(ggplot(df_Sites1, aes(x, y))+geom_point(aes(colour = df_Sites1$Ile))+geom_text(data=df_Sites1, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df_arrows1, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows1*1.1), aes(x, y, label = rownames(df_arrows1)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - All Environmental parameters", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites1, aes(x, y))+geom_point(aes(colour = df_Sites1$Ile, size = 2))+geom_segment(data = df_arrows1, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows1*1.1), aes(x, y, label = rownames(df_arrows1)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - All Environmental parameters", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites2, aes(x, y))+geom_point(aes(colour = df_Sites2$Ile))+geom_text(data=df_Sites2, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df_arrows2, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows2*1.1), aes(x, y, label = rownames(df_arrows2)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Primary physicochemical variables", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites2, aes(x, y))+geom_point(aes(colour = df_Sites2$Ile, size = 2))+geom_segment(data = df_arrows2, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows2*1.1), aes(x, y, label = rownames(df_arrows2)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Primary physicochemical variables", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites3, aes(x, y))+geom_point(aes(colour = df_Sites3$Ile))+geom_text(data=df_Sites3, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df_arrows3, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows3*1.1), aes(x, y, label = rownames(df_arrows3)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Micronutrients", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites3, aes(x, y))+geom_point(aes(colour = df_Sites3$Ile, size = 2))+geom_segment(data = df_arrows3, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows3*1.1), aes(x, y, label = rownames(df_arrows3)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Micronutrients", x= "CCA1", y ="CCA2", fill = "legend"))



plot(ggplot(df_Sites4, aes(x, y))+geom_point(aes(colour = df_Sites4$Ile))+geom_text(data=df_Sites4, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df_arrows4, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows4*1.1), aes(x, y, label = rownames(df_arrows4)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Trace Minerals", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites4, aes(x, y))+geom_point(aes(colour = df_Sites4$Ile, size = 2))+geom_segment(data = df_arrows4, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows4*1.1), aes(x, y, label = rownames(df_arrows4)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Trace Minerals", x= "CCA1", y ="CCA2", fill = "legend"))
dev.off()

# With filtering FC
# CCA
## All Parameters
cca2.1 <- cca(data.2 ~., data = env)
scrs2.1 <- scores(cca2.1, display=c("sp","wa","lc","bp","cn"))
#scrs2.1 <- scores(cca2.1, display=c("sp","bp"))
#attributes(scrs2.1)

df2_Sites1 <- data.frame(scrs2.1$sites)
colnames(df2_Sites1) <- c("x","y")
df2_Sites1$Samples <- rownames(df2_Sites1)
df2_Sites1$Ile <- substr(rownames(data.2), 1, 3)

multiplier2.1 <- vegan:::ordiArrowMul(scrs2.1$biplot)
df2_arrows1 <- as.data.frame(scrs2.1$biplot*multiplier2.1)
colnames(df2_arrows1)<-c("x","y")

## Primary physicochemical variables
cca2.2 <- cca(data.2 ~ Temp+Chla+PAR+Primary_Production_g_par_m3+O2+Salinity+PIC+POC, data = env)
scrs2.2 <- scores(cca2.2, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs2.2)

df2_Sites2 <- data.frame(scrs2.2$sites)
colnames(df2_Sites2) <- c("x","y")
df2_Sites2$Samples <- rownames(df2_Sites2)
df2_Sites2$Ile <- substr(rownames(data.2), 1, 3)

multiplier2.2 <- vegan:::ordiArrowMul(scrs2.2$biplot)
df2_arrows2 <- as.data.frame(scrs2.2$biplot*multiplier2.2)
colnames(df2_arrows2)<-c("x","y")

## Micronutrients
cca2.3 <- cca(data.2 ~ Nitrates_uM+Nitrites_uM+Phosphates_uM+Silice_uM+log_Fe_nM, data = env)
scrs2.3 <- scores(cca2.3, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs2.3)

df2_Sites3 <- data.frame(scrs2.3$sites)
colnames(df2_Sites3) <- c("x","y")
df2_Sites3$Samples <- rownames(df2_Sites3)
df2_Sites3$Ile <- substr(rownames(data.2), 1, 3)

multiplier2.3 <- vegan:::ordiArrowMul(scrs2.3$biplot)
df2_arrows3 <- as.data.frame(scrs2.3$biplot*multiplier2.3)
colnames(df2_arrows3)<-c("x","y")

## Trace Minerals
cca2.4 <- cca(data.2 ~ Zinc_nM+Cadmium_pM+Nickel_nM+Cuivre_nM+Plomb_pM+Magnesium_pM+Cobalt_pM, data = env)
scrs2.4 <- scores(cca2.4, display=c("sp","wa","lc","bp","cn"))
#attributes(scrs2.4)

df2_Sites4 <- data.frame(scrs2.4$sites)
colnames(df2_Sites4) <- c("x","y")
df2_Sites4$Samples <- rownames(df2_Sites4)
df2_Sites4$Ile <- substr(rownames(data.2), 1, 3)

multiplier2.4 <- vegan:::ordiArrowMul(scrs2.4$biplot)
df2_arrows4 <- as.data.frame(scrs2.4$biplot*multiplier2.4)
colnames(df2_arrows4)<-c("x","y")

# Graphs
pdf(file = "CCA-Environmental-parameters-Cladocopium-C1-Pocillopora-Graphs-With-Filter-By-Islands.pdf", width = 14, heigh = 12)
plot(ggplot(df2_Sites1, aes(x, y))+geom_point(aes(colour = df2_Sites1$Ile))+geom_text(data=df2_Sites1, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df2_arrows1, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows1*1.1), aes(x, y, label = rownames(df2_arrows1)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - All Environmental parameters - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites1, aes(x, y))+geom_point(aes(colour = df2_Sites1$Ile))+geom_segment(data = df2_arrows1, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows1*1.1), aes(x, y, label = rownames(df2_arrows1)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - All Environmental parameters - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites2, aes(x, y))+geom_point(aes(colour = df2_Sites2$Ile))+geom_text(data=df2_Sites2, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df2_arrows2, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows2*1.1), aes(x, y, label = rownames(df2_arrows2)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Primary physicochemical variables - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites2, aes(x, y))+geom_point(aes(colour = df2_Sites2$Ile))+geom_segment(data = df2_arrows2, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows2*1.1), aes(x, y, label = rownames(df2_arrows2)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Primary physicochemical variables - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites3, aes(x, y))+geom_point(aes(colour = df2_Sites3$Ile))+geom_text(data=df2_Sites3, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df2_arrows3, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows3*1.1), aes(x, y, label = rownames(df2_arrows3)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Micronutrients - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites3, aes(x, y))+geom_point(aes(colour = df2_Sites3$Ile))+geom_segment(data = df2_arrows3, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows3*1.1), aes(x, y, label = rownames(df2_arrows3)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Micronutrients - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites4, aes(x, y))+geom_point(aes(colour = df2_Sites4$Ile))+geom_text(data=df2_Sites4, aes(x, y,label = Samples, colour = Ile))+geom_segment(data = df2_arrows4, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows4*1.1), aes(x, y, label = rownames(df2_arrows4)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Trace Minerals - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df2_Sites4, aes(x, y))+geom_point(aes(colour = df2_Sites4$Ile))+geom_segment(data = df2_arrows4, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df2_arrows4*1.1), aes(x, y, label = rownames(df2_arrows4)))+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Trace Minerals - log2(FC) >= 2 & log2(FC) <= -2 - pvalue < 0.001", x= "CCA1", y ="CCA2", fill = "legend"))
dev.off()





















# Graphs
colors <- c("#F46D43", "#FEE08B", "black", "#ABDDA4", "#3288BD", "#5E4FA2")#, "#5E4FA2")
pdf(file = "CCA-Environmental-parameters-Cladocopium-C1-Pocillopora-Graphs-By-Islands2.pdf",  width = 18, heigh = 18)
plot(ggplot(df_Sites1, aes(x, y))+geom_point(aes(colour = df_Sites1$Ile, size = 2))+geom_segment(data = df_arrows1, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows1*1.1), aes(x, y, label = rownames(df_arrows1)), size = 7)+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - All Environmental parameters", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites2, aes(x, y))+geom_point(aes(colour = df_Sites2$Ile, size = 2))+geom_segment(data = df_arrows2, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows2*1.1), aes(x, y, label = rownames(df_arrows2)), size = 7)+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Primary physicochemical variables", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites3, aes(x, y))+geom_point(aes(colour = df_Sites3$Ile, size = 2))+geom_segment(data = df_arrows3, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows3*1.1), aes(x, y, label = rownames(df_arrows3)), size = 7)+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Micronutrients", x= "CCA1", y ="CCA2", fill = "legend"))

plot(ggplot(df_Sites4, aes(x, y))+geom_point(aes(colour = df_Sites4$Ile, size = 2))+geom_segment(data = df_arrows4, aes(x = 0, y = 0, xend = x, yend = y), arrow = arrow(length = unit(0.2, "cm")))+geom_text(data = as.data.frame(df_arrows4*1.1), aes(x, y, label = rownames(df_arrows4)), size = 7)+theme_bw()+scale_color_manual(values = colors)+labs(title = "CCA DEG from Pocillopora / Cladocopium C1 - TPM - Trace Minerals", x= "CCA1", y ="CCA2", fill = "legend"))
dev.off()
```

