---
title: "Cladocopium C1 DEG - DualT - Functional Analyses - 18.11.2020"
author: "Julie LÃª-Hoang"
contact: "lehoang.julie@hotmail.fr"
---

# 1. Load data and packages

## 1.1. Load Packages

```{r}
library("ggplot2")
library("stringr")
library("RColorBrewer")
library("dplyr")
library("reshape2")
library("ggrepel")
library("FactoMineR")
library("pheatmap")
library("htmlwidgets")
library("stringi")
library("DESeq2")
library("limma")
library("gplots")
library("venneuler")
library("VennDiagram")
library("eulerr")
library("factoextra")
library("treemap")
library("GOplot")
library("vegan")
library("seqinr")
library(openxlsx)
library(MASS)
library("BiocManager")
library("goseq")
library("GO.db")
library("geneLenDataBase")
library("qvalue")
library("topGO")
library(BiocManager)
library("Rgraphviz")
library("topGO")
library("hgu95av2.db")
```

## 1.2. Load DEG - Data 

```{r}
DEG <- read.table("DEG-Cladocopium-C1-Pocillopora-Groups-Comparisons.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
DEG$Comparisons <- rep("Cladocopium-Groups", nrow(DEG))

DEG.Specific.Analyses <- DEG
colnames(DEG.Specific.Analyses) <- c("Genes", "baseMean", "log2FoldChange", "lfcSE", "pvalue","padj", "Condition", "Comparisons")

SA <- data.frame(DEG.Specific.Analyses)
rownames(SA) <- c(1:nrow(SA))

write.table(SA, file ="DEG-Cladocopium-C1-Groups-Pocillopora-Analyses.txt", quote = FALSE, sep = "\t", row.names = FALSE)
```

## 1.3. Count Reads

```{r}
mapC1 <- read.table("DualT-Statistics-Alignment-Results-DualT-Samples-POC-C1-CDS-D2-transcriptome-98i-80l-Without-Rmdup.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
colnames(mapC1) <- c("QuerySeqID","Coverage","Identity","Count.Read","Length","Origin")
mapC1 <- mapC1[-which(mapC1[,1]=="QuerySeqID"),]

mapC1 <- mapC1[as.numeric(mapC1$Identity) != 0.0,]
mapC1$Transcriptome<-sub("_.*","",mapC1$QuerySeqID)
mapC1$Clade<-substr(mapC1$QuerySeqID,2,2)
mapC1$Identity <- as.numeric(mapC1$Identity)
mapC1$Coverage <- as.numeric(mapC1$Coverage)
mapC1$Length <- as.numeric(mapC1$Length)
mapC1$Count.Read <- as.numeric(mapC1$Count.Read)
mapC1 <- mapC1[-grep("D2", mapC1$Transcriptome),]

lineage1 <- str_split(mapC1$QuerySeqID,"_",3)
lineage2 <- c()
myColors <- c()
for (i in 1:length(lineage1)){
  lineage2[i] = lineage1[[i]][2]
  #myColors[i] = as.character(colorsTab[as.numeric(grep(sub("TaraDB-", "",lineage1[[i]][2]), colorsTab$myGroups)),2])
}

mapC1$Lineage <- sub("TaraDB-", "",lineage2)
mapC1$Lineage<-factor(mapC1$Lineage,levels=c("Symbiodinium","Suessiales","Dinophyceae", "Ciliophora","Alveolata","Haptophyceae","Cryptophyta","Stramenopiles","Chlorophyta","Viridiplantae","Rhizaria","Bilateria","Amoebozoa","Cnidaria","Metazoa","Fungi","Eukaryota","Bacteria","root","cellularOrganisms","NoMatch","ID-NOT-FOUND"))
mapC1$Island <- substr(mapC1$Origin, 1, 3)
mapC1 <- mapC1[-grep("I01S01", mapC1$Origin),]
mapC1 <- mapC1[-grep("I01S02C010", mapC1$Origin),]
mapC1 <- mapC1[-grep("I01S03C001", mapC1$Origin),]
mapC1 <- mapC1[-grep("I02S01", mapC1$Origin),]
mapC1 <- mapC1[-grep("I02S02C006", mapC1$Origin),]
mapC1 <- mapC1[-grep("I02S03", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C011", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C012", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C014", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C015", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C016", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C020", mapC1$Origin),]
mapC1 <- mapC1[-grep("I15S02C010", mapC1$Origin),]
mapC1 <- mapC1[-grep("I10S01C006", mapC1$Origin),]
mapC1 <- mapC1[-grep("I05S02C002", mapC1$Origin),]

remove(lineage1)
remove(lineage2)

mapC1 <- mapC1[order(mapC1$QuerySeqID),]

dataAll <- acast(mapC1, QuerySeqID~Origin, fill=0, value.var="Count.Read")


# TPM 
lengths <- mapC1[duplicated(mapC1$QuerySeqID)==F, 5]
tabTPM <- t(t(dataAll/(lengths/1000))/colSums(dataAll/(lengths/1000)))*1000000

write.table(tabTPM, file = "TPM-Pocillopora-Cladocopium-C1-CDS.txt", quote = FALSE, sep = "\t", row.names = TRUE, col.names = TRUE)

write.table(dataAll, file = "Raw-Count-Pocillopora-Cladocopium-C1-CDS.txt", quote = FALSE, sep = "\t", row.names = TRUE, col.names = TRUE)

# Defined Population Structure
G1 <- c("I04S01C001POC", "I04S01C006POC", "I04S01C010POC", "I04S02C001POC", "I04S02C006POC", "I04S02C010POC", "I04S03C001POC", "I04S03C006POC", "I04S03C010POC", "I04S04C001POC", "I04S04C006POC", "I04S04C010POC")

G2 <- c("I05S01C010POC", "I05S02C006POC", "I05S02C010POC", "I06S01C001POC", "I06S01C006POC", "I06S02C001POC", "I06S02C006POC", "I06S03C001POC", "I06S03C006POC", "I06S03C010POC", "I07S03C010POC")

G3 <- c("I05S03C001POC", "I05S03C006POC", "I07S01C001POC", "I07S01C006POC", "I07S01C010POC", "I07S02C001POC", "I07S02C006POC", "I07S03C001POC", "I07S03C006POC", "I08S01C001POC", "I08S01C010POC", "I08S02C001POC", "I08S03C001POC", "I08S03C010POC", "I09S01C001POC", "I09S02C001POC", "I09S02C006POC", "I09S02C010POC", "I09S03C006POC", "I10S02C010POC", "I15S01C006POC", "I15S01C010POC", "I15S02C001POC", "I15S03C001POC", "I15S03C010POC")

G4 <- c("I05S01C001POC", "I05S01C006POC", "I05S03C010POC", "I06S01C010POC", "I06S02C010POC")

G5 <- c("I01S02C001POC", "I01S02C006POC", "I01S03C006POC", "I01S03C010POC", "I02S02C001POC", "I02S02C010POC", "I03S01C013POC", "I03S01C017POC", "I03S01C018POC", "I03S01C019POC", "I07S02C010POC", "I08S01C006POC", "I08S02C006POC", "I08S02C010POC", "I08S03C006POC", "I09S01C006POC", "I09S01C010POC", "I09S03C001POC", "I09S03C010POC", "I10S01C001POC", "I10S01C010POC", "I10S02C001POC", "I10S02C006POC", "I10S03C001POC", "I10S03C006POC", "I10S03C010POC", "I15S01C001POC", "I15S02C006POC", "I15S03C006POC")

mapC1$Cladocopium <- rep("Not.Assigned", nrow(mapC1))
mapC1[mapC1$Origin%in%G1, c("Cladocopium")] <- "G1"
mapC1[mapC1$Origin%in%G2, c("Cladocopium")] <- "G2"
mapC1[mapC1$Origin%in%G3, c("Cladocopium")] <- "G3"
mapC1[mapC1$Origin%in%G4, c("Cladocopium")] <- "G4"
mapC1[mapC1$Origin%in%G5, c("Cladocopium")] <- "G5"

# Defined Population Structure - Corals
GNA <- c("I09S03C010POC", "I07S03C001POC", "I09S02C001POC")

SVD1 <- c("I05S01C010POC", "I05S02C002POC", "I05S02C006POC", "I05S02C010POC", "I06S01C001POC", "I06S01C006POC", "I06S02C001POC", "I06S02C006POC", "I06S03C001POC", "I06S03C006POC", "I06S03C010POC", "I07S03C010POC")

SVD2 <- c("I15S03C006POC", "I15S01C001POC", "I07S02C010POC", "I08S01C006POC", "I08S02C006POC", "I08S02C010POC", "I10S01C010POC", "I10S01C001POC", "I10S02C001POC", "I10S02C006POC", "I10S03C001POC", "I10S03C006POC", "I10S03C010POC", "I08S03C006POC", "I09S01C006POC", "I09S01C010POC", "I09S03C001POC", "I15S02C006POC")

SVD3 <- c("I15S03C001POC", "I15S03C010POC", "I15S02C001POC", "I15S02C010POC", "I15S01C006POC", "I15S01C010POC", "I08S01C001POC", "I08S01C010POC", "I08S02C001POC", "I10S01C006POC", "I10S02C010POC", "I09S02C006POC", "I09S02C010POC", "I08S03C001POC", "I08S03C010POC", "I09S01C001POC", "I09S03C006POC")

SVD4 <- c("I01S01C001POC", "I01S01C006POC", "I01S01C010POC", "I01S02C001POC", "I01S02C006POC", "I01S02C010POC", "I02S01C001POC", "I02S01C006POC", "I02S01C010POC", "I02S02C001POC", "I02S02C006POC", "I02S02C010POC", "I02S03C001POC", "I02S03C006POC", "I02S03C010POC", "I01S03C006POC", "I01S03C010POC", "I03S01C011POC", "I03S01C012POC", "I03S01C013POC", "I03S01C014POC", "I03S01C015POC", "I03S01C016POC", "I03S01C017POC", "I03S01C018POC", "I03S01C019POC", "I03S01C20POC","I05S01C001POC", "I05S01C006POC", "I05S03C010POC", "I06S01C010POC", "I06S02C010POC")

SVD5 <- c("I04S01C001POC", "I04S01C006POC", "I04S01C010POC", "I04S02C001POC", "I04S02C006POC", "I04S02C010POC", "I04S03C001POC", "I04S03C006POC", "I04S03C010POC", "I04S04C001POC", "I04S04C006POC", "I04S04C010POC", "I07S01C001POC", "I07S01C006POC", "I07S01C010POC", "I07S02C001POC", "I07S02C006POC", "I07S03C006POC", "I05S03C001POC", "I05S03C006POC")

mapC1$Host.Pop <- rep("Host.Pop", nrow(mapC1))
mapC1[mapC1$Origin%in%SVD1, c("Host.Pop")] <- "SVD1"
mapC1[mapC1$Origin%in%SVD2, c("Host.Pop")] <- "SVD2"
mapC1[mapC1$Origin%in%SVD3, c("Host.Pop")] <- "SVD3"
mapC1[mapC1$Origin%in%SVD4, c("Host.Pop")] <- "SVD4"
mapC1[mapC1$Origin%in%SVD5, c("Host.Pop")] <- "SVD5"
mapC1[mapC1$Origin%in%GNA, c("Host.Pop")] <- "Not.Assigned"

test <- unique(mapC1[,c("Origin","Host.Pop")])
test2 <- unique(mapC1[,c("Origin","Cladocopium")])
```

## 1.4. Functions

```{r}
Func <- read.table("Annotation_C1_Genome.txt",sep="\t",h=T, quote = "")

# Modified MapC1
lineage1 <- str_split(mapC1$QuerySeqID,"_",3)
lineage2 <- c()
for (i in 1:length(lineage1)){
  lineage2[i] = lineage1[[i]][3]
}
mapC1$Gene.model <- paste("SymbC1.", lineage2, sep ="")

# Modified SA
lineage1 <- str_split(SA$Genes,"_",3)
lineage2 <- c()
for (i in 1:length(lineage1)){
  lineage2[i] = lineage1[[i]][3]
}
SA$Gene.model <- paste("SymbC1.", lineage2, sep ="")

# Link Between Data Frames
myDatalink.SA <- merge(SA, Func, by="Gene.model", all.x=T, sort = FALSE)
myDatalink.mapC1 <- merge(mapC1, Func, by="Gene.model", all.x=T, sort = FALSE)

write.table(myDatalink.SA, file ="Functions-DEG-C1-Groups-Merged.txt", quote = FALSE, sep = "\t", row.names = FALSE)

# Remove NA in pvalue
myDatalink.SA.1 <- myDatalink.SA[!is.na(myDatalink.SA$padj),]
# Filter p.value
myDatalink.SA.Filtered <- myDatalink.SA.1[myDatalink.SA.1$padj < 0.001,]

write.table(myDatalink.SA.Filtered, file ="Functions-DEG-C1-Groups-Merged-Filtered.txt", quote = FALSE, sep = "\t", row.names = FALSE)
```

# 2. KO

```{r}
# Separate KO and PFAM in Excel to have easier way to play with data
myNewDatalink.SA.Filtered <- read.table("Functions-DEG-C1-Groups-Merged-Filtered-New.txt",sep="\t",h=T, quote = "", stringsAsFactors = FALSE, na.strings = c("", "NA"))

#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("KEGGprofile")
#library("KEGGprofile")

#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("KEGGREST")
library("KEGGREST")

# smin Code for Breviolum minutum
#keggList("smin")

head(myNewDatalink.SA.Filtered)

# KEGG1
KEGG1 <- c(myNewDatalink.SA.Filtered$KEGG1)
KEGG1.Bis <- c(na.omit(unique(KEGG1[order(KEGG1, decreasing = FALSE)])))

KOlist.name1 <- c()
KOlist.def1 <- c()
for(i in 1:length(KEGG1.Bis)){
  
  if(length(keggFind("ko", KEGG1.Bis[i])) != 0){
    KOlist.name1[i] <- KEGG1.Bis[i]
    KOlist.def1[i] <- keggFind("ko", KEGG1.Bis[i])
  }
  else{
    KOlist.name1[i] <- KEGG1.Bis[i]
    KOlist.def1[i] <- "NA"   
  }
}

# KEGG2
KEGG2 <- c(myNewDatalink.SA.Filtered$KEGG2)
KEGG2.Bis <- c(na.omit(unique(KEGG2[order(KEGG2, decreasing = FALSE)])))

KOlist.name2 <- c()
KOlist.def2 <- c()
for(i in 1:length(KEGG2.Bis)){
  
  if(length(keggFind("ko", KEGG2.Bis[i])) != 0){
    KOlist.name2[i] <- KEGG2.Bis[i]
    KOlist.def2[i] <- keggFind("ko", KEGG2.Bis[i])
  }
  else{
    KOlist.name2[i] <- KEGG2.Bis[i]
    KOlist.def2[i] <- "NA"   
  }
}

NA.val <- c(NA, NA)
names(NA.val) <- c("KEGG1", "KEGG1.Def")

KEGG1.List <- data.frame(KEGG1 = KOlist.name1, KEGG1.Def = KOlist.def1)
KEGG1.List <- rbind(KEGG1.List, NA.val)
KEGG2.List <- data.frame(KEGG2 = KOlist.name2, KEGG2.Def = KOlist.def2)
KEGG2.List <- rbind(KEGG2.List,NA.val)

Merge.KEGG1 <- merge(KEGG1.List, myNewDatalink.SA.Filtered, by="KEGG1", all.x=T, sort = FALSE)
Merge.KEGG2 <- merge(KEGG2.List, Merge.KEGG1, by="KEGG2", all.x=T, sort = FALSE)
```

# 3. PFAM

```{r}
PFAM <- read.table("pfamA-New.txt", sep="\t", h=F, quote = "", stringsAsFactors = FALSE, na.strings = c("", "NA"))

# PFAM1
colnames(PFAM) <- c("PFAM1", "PFAM1.Sum", "PFAM1.Def")
Merge.PFAM1 <- merge(Merge.KEGG2, PFAM, by="PFAM1", all.x=T, sort = FALSE)

# PFAM2
colnames(PFAM) <- c("PFAM2", "PFAM2.Sum", "PFAM2.Def")
Merge.PFAM2 <- merge(Merge.PFAM1, PFAM, by="PFAM2", all.x=T, sort = FALSE)

# PFAM3
colnames(PFAM) <- c("PFAM3", "PFAM3.Sum", "PFAM3.Def")
Merge.PFAM3 <- merge(Merge.PFAM2, PFAM, by="PFAM3", all.x=T, sort = FALSE)

# PFAM4
colnames(PFAM) <- c("PFAM4", "PFAM4.Sum", "PFAM4.Def")
Merge.PFAM4 <- merge(Merge.PFAM3, PFAM, by="PFAM4", all.x=T, sort = FALSE)

# PFAM5
colnames(PFAM) <- c("PFAM5", "PFAM5.Sum", "PFAM5.Def")
Merge.PFAM5 <- merge(Merge.PFAM4, PFAM, by="PFAM5", all.x=T, sort = FALSE)

# PFAM6
colnames(PFAM) <- c("PFAM6", "PFAM6.Sum", "PFAM6.Def")
Merge.PFAM6 <- merge(Merge.PFAM5, PFAM, by="PFAM6", all.x=T, sort = FALSE)

# PFAM7
colnames(PFAM) <- c("PFAM7", "PFAM7.Sum", "PFAM7.Def")
Merge.PFAM7 <- merge(Merge.PFAM6, PFAM, by="PFAM7", all.x=T, sort = FALSE)

# PFAM8
colnames(PFAM) <- c("PFAM8", "PFAM8.Sum", "PFAM8.Def")
Merge.PFAM8 <- merge(Merge.PFAM7, PFAM, by="PFAM8", all.x=T, sort = FALSE)

# PFAM9
colnames(PFAM) <- c("PFAM9", "PFAM9.Sum", "PFAM9.Def")
Merge.PFAM9 <- merge(Merge.PFAM8, PFAM, by="PFAM9", all.x=T, sort = FALSE)

# PFAM10
colnames(PFAM) <- c("PFAM10", "PFAM10.Sum", "PFAM10.Def")
Merge.PFAM10 <- merge(Merge.PFAM9, PFAM, by="PFAM10", all.x=T, sort = FALSE)

myData <- data.frame(Merge.PFAM10$Gene.model, Merge.PFAM10$Genes, Merge.PFAM10$baseMean, Merge.PFAM10$log2FoldChange, Merge.PFAM10$lfcSE, 
                     Merge.PFAM10$pvalue, Merge.PFAM10$padj, Merge.PFAM10$Condition, Merge.PFAM10$Comparisons,
                     Merge.PFAM10$UniProt.identifier..top.hit.and.or.top.hit.with.annotated.GO., 
                     Merge.PFAM10$Description.s..of.hit.s., Merge.PFAM10$E.value, Merge.PFAM10$Gene.Ontology..GO..term,
                     Merge.PFAM10$PFAM1, Merge.PFAM10$PFAM1.Sum, Merge.PFAM10$PFAM1.Def,
                     Merge.PFAM10$PFAM2, Merge.PFAM10$PFAM2.Sum, Merge.PFAM10$PFAM2.Def,
                     Merge.PFAM10$PFAM3, Merge.PFAM10$PFAM3.Sum, Merge.PFAM10$PFAM3.Def,
                     Merge.PFAM10$PFAM4, Merge.PFAM10$PFAM4.Sum, Merge.PFAM10$PFAM4.Def,
                     Merge.PFAM10$PFAM5, Merge.PFAM10$PFAM5.Sum, Merge.PFAM10$PFAM5.Def,
                     Merge.PFAM10$PFAM6, Merge.PFAM10$PFAM6.Sum, Merge.PFAM10$PFAM6.Def,
                     Merge.PFAM10$PFAM7, Merge.PFAM10$PFAM7.Sum, Merge.PFAM10$PFAM7.Def,
                     Merge.PFAM10$PFAM8, Merge.PFAM10$PFAM8.Sum, Merge.PFAM10$PFAM8.Def,
                     Merge.PFAM10$PFAM9, Merge.PFAM10$PFAM9.Sum, Merge.PFAM10$PFAM9.Def,
                     Merge.PFAM10$PFAM10, Merge.PFAM10$PFAM10.Sum, Merge.PFAM10$PFAM10.Def,
                     Merge.PFAM10$KEGG1, Merge.PFAM10$KEGG1.Def, Merge.PFAM10$KEGG2, Merge.PFAM10$KEGG2.Def)

colnames(myData) <- sub("Merge.PFAM10.","", colnames(myData))

write.table(myData, file ="Functions-DEG-C1-Groups-Merged-Filtered-KO-PFAM.txt", quote = FALSE, sep = "\t", row.names = FALSE)
```

