---
title: "Cladocopium C1 DEG - DualT - Functional Analyses - 18.11.2020"
author: "Julie LÃª-Hoang"
contact: "lehoang.julie@hotmail.fr"
---

# 1. Load data and packages

## 1.1. Load Packages

```{r}
library("ggplot2")
library("stringr")
library("RColorBrewer")
library("dplyr")
library("reshape2")
library("ggrepel")
library("FactoMineR")
library("pheatmap")
library("htmlwidgets")
library("stringi")
library("DESeq2")
library("limma")
library("gplots")
library("venneuler")
library("VennDiagram")
library("eulerr")
library("factoextra")
library("treemap")
library("GOplot")
library("vegan")
library("seqinr")
library(openxlsx)
library(MASS)
library("BiocManager")
library("goseq")
library("GO.db")
library("geneLenDataBase")
library("qvalue")
library("topGO")
library(BiocManager)
library("Rgraphviz")
library("topGO")
library("hgu95av2.db")
```

## 1.2. Load DEG - Data 

```{r}
DEG <- read.table("DEG-Cladocopium-C1-Pocillopora-Groups-Comparisons-Filtered.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)

DEG.Specific.Analyses <- DEG
colnames(DEG) <- c("Genes", "baseMean", "log2FoldChange", "lfcSE", "pvalue","padj", "Condition")

SA <- data.frame(DEG.Specific.Analyses)
rownames(SA) <- c(1:nrow(SA))
```

## 1.3. Count Reads

```{r}
mapC1 <- read.table("DualT-Statistics-Alignment-Results-DualT-Samples-POC-C1-CDS-D2-transcriptome-98i-80l-Without-Rmdup.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
colnames(mapC1) <- c("QuerySeqID","Coverage","Identity","Count.Read","Length","Origin")
mapC1 <- mapC1[-which(mapC1[,1]=="QuerySeqID"),]

mapC1 <- mapC1[as.numeric(mapC1$Identity) != 0.0,]
mapC1$Transcriptome<-sub("_.*","",mapC1$QuerySeqID)
mapC1$Clade<-substr(mapC1$QuerySeqID,2,2)
mapC1$Identity <- as.numeric(mapC1$Identity)
mapC1$Coverage <- as.numeric(mapC1$Coverage)
mapC1$Length <- as.numeric(mapC1$Length)
mapC1$Count.Read <- as.numeric(mapC1$Count.Read)
mapC1 <- mapC1[-grep("D2", mapC1$Transcriptome),]

lineage1 <- str_split(mapC1$QuerySeqID,"_",3)
lineage2 <- c()
myColors <- c()
for (i in 1:length(lineage1)){
  lineage2[i] = lineage1[[i]][2]
  #myColors[i] = as.character(colorsTab[as.numeric(grep(sub("TaraDB-", "",lineage1[[i]][2]), colorsTab$myGroups)),2])
}

mapC1$Lineage <- sub("TaraDB-", "",lineage2)
mapC1$Lineage<-factor(mapC1$Lineage,levels=c("Symbiodinium","Suessiales","Dinophyceae", "Ciliophora","Alveolata","Haptophyceae","Cryptophyta","Stramenopiles","Chlorophyta","Viridiplantae","Rhizaria","Bilateria","Amoebozoa","Cnidaria","Metazoa","Fungi","Eukaryota","Bacteria","root","cellularOrganisms","NoMatch","ID-NOT-FOUND"))
mapC1$Island <- substr(mapC1$Origin, 1, 3)
mapC1 <- mapC1[-grep("I01S01", mapC1$Origin),]
mapC1 <- mapC1[-grep("I01S02C010", mapC1$Origin),]
mapC1 <- mapC1[-grep("I01S03C001", mapC1$Origin),]
mapC1 <- mapC1[-grep("I02S01", mapC1$Origin),]
mapC1 <- mapC1[-grep("I02S02C006", mapC1$Origin),]
mapC1 <- mapC1[-grep("I02S03", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C011", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C012", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C014", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C015", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C016", mapC1$Origin),]
mapC1 <- mapC1[-grep("I03S01C020", mapC1$Origin),]
mapC1 <- mapC1[-grep("I15S02C010", mapC1$Origin),]
mapC1 <- mapC1[-grep("I10S01C006", mapC1$Origin),]
mapC1 <- mapC1[-grep("I05S02C002", mapC1$Origin),]

remove(lineage1)
remove(lineage2)

mapC1 <- mapC1[order(mapC1$QuerySeqID),]

dataAll <- acast(mapC1, QuerySeqID~Origin, fill=0, value.var="Count.Read")


# TPM 
lengths <- mapC1[duplicated(mapC1$QuerySeqID)==F, 5]
tabTPM <- t(t(dataAll/(lengths/1000))/colSums(dataAll/(lengths/1000)))*1000000

write.table(tabTPM, file = "TPM-Pocillopora-Cladocopium-C1-CDS.txt", quote = FALSE, sep = "\t", row.names = TRUE, col.names = TRUE)

write.table(dataAll, file = "Raw-Count-Pocillopora-Cladocopium-C1-CDS.txt", quote = FALSE, sep = "\t", row.names = TRUE, col.names = TRUE)

# Defined Population Structure
G1 <- c("I04S01C001POC", "I04S01C006POC", "I04S01C010POC", "I04S02C001POC", "I04S02C006POC", "I04S02C010POC", "I04S03C001POC", "I04S03C006POC", "I04S03C010POC", "I04S04C001POC", "I04S04C006POC", "I04S04C010POC")

G2 <- c("I05S01C010POC", "I05S02C006POC", "I05S02C010POC", "I06S01C001POC", "I06S01C006POC", "I06S02C001POC", "I06S02C006POC", "I06S03C001POC", "I06S03C006POC", "I06S03C010POC", "I07S03C010POC")

G3 <- c("I05S03C001POC", "I05S03C006POC", "I07S01C001POC", "I07S01C006POC", "I07S01C010POC", "I07S02C001POC", "I07S02C006POC", "I07S03C001POC", "I07S03C006POC", "I08S01C001POC", "I08S01C010POC", "I08S02C001POC", "I08S03C001POC", "I08S03C010POC", "I09S01C001POC", "I09S02C001POC", "I09S02C006POC", "I09S02C010POC", "I09S03C006POC", "I10S02C010POC", "I15S01C006POC", "I15S01C010POC", "I15S02C001POC", "I15S03C001POC", "I15S03C010POC")

G4 <- c("I05S01C001POC", "I05S01C006POC", "I05S03C010POC", "I06S01C010POC", "I06S02C010POC")

G5 <- c("I01S02C001POC", "I01S02C006POC", "I01S03C006POC", "I01S03C010POC", "I02S02C001POC", "I02S02C010POC", "I03S01C013POC", "I03S01C017POC", "I03S01C018POC", "I03S01C019POC", "I07S02C010POC", "I08S01C006POC", "I08S02C006POC", "I08S02C010POC", "I08S03C006POC", "I09S01C006POC", "I09S01C010POC", "I09S03C001POC", "I09S03C010POC", "I10S01C001POC", "I10S01C010POC", "I10S02C001POC", "I10S02C006POC", "I10S03C001POC", "I10S03C006POC", "I10S03C010POC", "I15S01C001POC", "I15S02C006POC", "I15S03C006POC")

mapC1$Cladocopium <- rep("Not.Assigned", nrow(mapC1))
mapC1[mapC1$Origin%in%G1, c("Cladocopium")] <- "G1"
mapC1[mapC1$Origin%in%G2, c("Cladocopium")] <- "G2"
mapC1[mapC1$Origin%in%G3, c("Cladocopium")] <- "G3"
mapC1[mapC1$Origin%in%G4, c("Cladocopium")] <- "G4"
mapC1[mapC1$Origin%in%G5, c("Cladocopium")] <- "G5"

# Defined Population Structure - Corals
GNA <- c("I09S03C010POC", "I07S03C001POC", "I09S02C001POC")

SVD1 <- c("I05S01C010POC", "I05S02C002POC", "I05S02C006POC", "I05S02C010POC", "I06S01C001POC", "I06S01C006POC", "I06S02C001POC", "I06S02C006POC", "I06S03C001POC", "I06S03C006POC", "I06S03C010POC", "I07S03C010POC")

SVD2 <- c("I15S03C006POC", "I15S01C001POC", "I07S02C010POC", "I08S01C006POC", "I08S02C006POC", "I08S02C010POC", "I10S01C010POC", "I10S01C001POC", "I10S02C001POC", "I10S02C006POC", "I10S03C001POC", "I10S03C006POC", "I10S03C010POC", "I08S03C006POC", "I09S01C006POC", "I09S01C010POC", "I09S03C001POC", "I15S02C006POC")

SVD3 <- c("I15S03C001POC", "I15S03C010POC", "I15S02C001POC", "I15S02C010POC", "I15S01C006POC", "I15S01C010POC", "I08S01C001POC", "I08S01C010POC", "I08S02C001POC", "I10S01C006POC", "I10S02C010POC", "I09S02C006POC", "I09S02C010POC", "I08S03C001POC", "I08S03C010POC", "I09S01C001POC", "I09S03C006POC")

SVD4 <- c("I01S01C001POC", "I01S01C006POC", "I01S01C010POC", "I01S02C001POC", "I01S02C006POC", "I01S02C010POC", "I02S01C001POC", "I02S01C006POC", "I02S01C010POC", "I02S02C001POC", "I02S02C006POC", "I02S02C010POC", "I02S03C001POC", "I02S03C006POC", "I02S03C010POC", "I01S03C006POC", "I01S03C010POC", "I03S01C011POC", "I03S01C012POC", "I03S01C013POC", "I03S01C014POC", "I03S01C015POC", "I03S01C016POC", "I03S01C017POC", "I03S01C018POC", "I03S01C019POC", "I03S01C20POC","I05S01C001POC", "I05S01C006POC", "I05S03C010POC", "I06S01C010POC", "I06S02C010POC")

SVD5 <- c("I04S01C001POC", "I04S01C006POC", "I04S01C010POC", "I04S02C001POC", "I04S02C006POC", "I04S02C010POC", "I04S03C001POC", "I04S03C006POC", "I04S03C010POC", "I04S04C001POC", "I04S04C006POC", "I04S04C010POC", "I07S01C001POC", "I07S01C006POC", "I07S01C010POC", "I07S02C001POC", "I07S02C006POC", "I07S03C006POC", "I05S03C001POC", "I05S03C006POC")

mapC1$Host.Pop <- rep("Host.Pop", nrow(mapC1))
mapC1[mapC1$Origin%in%SVD1, c("Host.Pop")] <- "SVD1"
mapC1[mapC1$Origin%in%SVD2, c("Host.Pop")] <- "SVD2"
mapC1[mapC1$Origin%in%SVD3, c("Host.Pop")] <- "SVD3"
mapC1[mapC1$Origin%in%SVD4, c("Host.Pop")] <- "SVD4"
mapC1[mapC1$Origin%in%SVD5, c("Host.Pop")] <- "SVD5"
mapC1[mapC1$Origin%in%GNA, c("Host.Pop")] <- "Not.Assigned"

test <- unique(mapC1[,c("Origin","Host.Pop")])
test2 <- unique(mapC1[,c("Origin","Cladocopium")])
```

## 1.4. Functions

```{r}
Func <- read.table("Annotation_C1_Genome.txt",sep="\t",h=T, quote = "")

# Modified MapC1
lineage1 <- str_split(mapC1$QuerySeqID,"_",3)
lineage2 <- c()
for (i in 1:length(lineage1)){
  lineage2[i] = lineage1[[i]][3]
}
mapC1$Gene.model <- paste("SymbC1.", lineage2, sep ="")

# Modified SA
lineage1 <- str_split(SA$Genes,"_",3)
lineage2 <- c()
for (i in 1:length(lineage1)){
  lineage2[i] = lineage1[[i]][3]
}
SA$Gene.model <- paste("SymbC1.", lineage2, sep ="")

# Link Between Data Frames
myDatalink.SA <- merge(SA, Func, by="Gene.model", all.x=T, sort = FALSE)
myDatalink.mapC1 <- merge(mapC1, Func, by="Gene.model", all.x=T, sort = FALSE)

write.table(myDatalink.SA, file ="Functions-All-DEG-Cladocopium-C1-Pocillopora-Analyses-by-Cladocopium-Groups-Merged.txt", quote = FALSE, sep = "\t", row.names = FALSE)
```

# 2. topGO tables

## 2.1. Functions

```{r}
# Function TOPGO
topGO.fun <- function(Condition, topGOtable){
  AllmyGO.VF <- data.frame()
  for(i in 1:length(Condition)){
    topGOtable.select <- topGOtable[grep(Condition[i], topGOtable$Condition),]
   
    # Remove NA in pvalue
    topGOtable.select <- topGOtable.select[!is.na(topGOtable.select$padj),]

    # Filter p.value and defined gene list
    geneList <- topGOtable.select$padj
    names(geneList) <- topGOtable.select$Genes
    geneList <- na.omit(geneList)
    #sum(topDiffGenes(geneList))
    
    #GO database from DEG selected
    geneID2GO <- as.character(factor(topGOtable.select$Gene.Ontology..GO..term))
    geneID2GO <- sub("\"", "", geneID2GO)
    geneID2GO <- sub("\\\"", "", geneID2GO)
    names(geneID2GO) <- topGOtable.select$Genes
    
    # Molecular functions
    GOdataMF <- new("topGOdata", ontology = "MF", allGenes = geneList, geneSel = topDiffGenes, 
                    nodeSize = 1, annot = annFUN.gene2GO, gene2GO = geneID2GO)
    
    # Statistics tests
    resultFisherMF <- runTest(GOdataMF, algorithm = "classic", statistic = "fisher")
    resultKSMF <- runTest(GOdataMF, algorithm = "classic", statistic = "ks")
    resultKS.elimMF <- runTest(GOdataMF, algorithm = "elim", statistic = "ks")    
    allResMF <- GenTable(GOdataMF, classicFisher = resultFisherMF, classicKS = resultKSMF, 
                         elimKS = resultKS.elimMF, topNodes = resultKS.elimMF@geneData[4])
    allResMF$GOH <- rep("MF", nrow(allResMF))
    
    # Biological Process
    GOdataBP <- new("topGOdata", ontology = "BP", allGenes = geneList, geneSel = topDiffGenes, 
                    nodeSize = 1, annot = annFUN.gene2GO, gene2GO = geneID2GO)
    # Statistics tests
    resultFisherBP <- runTest(GOdataBP, algorithm = "classic", statistic = "fisher")
    resultKSBP <- runTest(GOdataBP, algorithm = "classic", statistic = "ks")
    resultKS.elimBP <- runTest(GOdataBP, algorithm = "elim", statistic = "ks")    
    allResBP <- GenTable(GOdataBP, classicFisher = resultFisherBP, classicKS = resultKSBP, elimKS =
                           resultKS.elimBP, topNodes = resultKS.elimBP@geneData[4])
    allResBP$GOH <- rep("BP", nrow(allResBP))
    
    # cellular component
    GOdataCC <- new("topGOdata", ontology = "CC", allGenes = geneList, geneSel = topDiffGenes, 
                    nodeSize = 1, annot = annFUN.gene2GO, gene2GO = geneID2GO)
    # Statistics tests
    resultFisherCC <- runTest(GOdataCC, algorithm = "classic", statistic = "fisher")
    resultKSCC <- runTest(GOdataCC, algorithm = "classic", statistic = "ks")
    resultKS.elimCC <- runTest(GOdataCC, algorithm = "elim", statistic = "ks")    
    allResCC <- GenTable(GOdataCC, classicFisher = resultFisherCC, classicKS = resultKSCC, 
                         elimKS = resultKS.elimCC, topNodes = resultKS.elimCC@geneData[4])
    allResCC$GOH <- rep("CC", nrow(allResCC))
    
    pdf(file = paste("Graphs-GO-Functions-DEG-Islands-Inside-",Condition[i],"-Cladocopium-C1-Pocillopora.pdf", sep = ""), height = 18)
      showSigOfNodes(GOdataMF, score(resultKS.elimMF), firstSigNodes = 10, useInfo = 'all')
      showSigOfNodes(GOdataBP, score(resultKS.elimBP), firstSigNodes = 10, useInfo = 'all')
      showSigOfNodes(GOdataCC, score(resultKS.elimCC), firstSigNodes = 10, useInfo = 'all')
    dev.off()
    
    AllmyGO <- rbind(allResMF, allResBP, allResCC)
    AllmyGO$Condition <- rep(Condition[i], nrow(AllmyGO))
    write.table(AllmyGO, file = paste("GO-Functions-DEG-Islands-Inside-",Condition[i],"-Cladocopium-C1-Pocillopora.txt", sep = ""), quote = FALSE, sep = "\t", row.names = FALSE)
    AllmyGO.VF <- rbind(AllmyGO.VF, AllmyGO)
  }
  AllmyGO.VF
}

# Create a Function to select DEGs from geneList
topDiffGenes <- function(x) {
  return(x < padj)
}

# Parameters to defined GO enrichment
padj <- 0.001 
```

## 2.2. GO

```{r}
topGOtableAll <- myDatalink.SA
topGOtableAll$Up.Down <- rep("A", nrow(topGOtableAll))
topGOtableAll$Up.Down[topGOtableAll$log2FoldChange >= 0] <- "Up"
topGOtableAll$Up.Down[topGOtableAll$log2FoldChange <= 0] <- "Down"
topGOtableAll$Condition <- paste(topGOtableAll$Condition, topGOtableAll$Up.Down, sep ="-")
ConditionAll.Bis <- unique(topGOtableAll$Condition)
ConditionAll <- ConditionAll.Bis[order(ConditionAll.Bis, decreasing = FALSE)]

topGOtableVF <- topGO.fun(Condition = ConditionAll, topGOtable = topGOtableAll)
```

## 2.6. Combo

```{r}
AllmyGOVF <- topGOtableVF
write.table(AllmyGOVF, file ="Functions-GO-DEG-C1-Analyses-Islands-Inside-Groups.txt", quote = FALSE, sep = "\t", row.names = FALSE)
```


# 3. topGO tables - Up and Down

## 3.1. Functions

```{r}
# Function TOPGO
topGO.fun.Up <- function(Condition, topGOtable, log2FC){
  AllmyGO.VF <- data.frame()
  for(i in 1:length(Condition)){
    topGOtable.select1 <- topGOtable[grep(Condition[i], topGOtable$Condition),]
   
    # Remove NA in pvalue
    topGOtable.select <- topGOtable.select1[!is.na(topGOtable.select1$padj),]
    topGOtable.select1$FoldChange <- 2^topGOtable.select1$log2FoldChange
    topGOtable.select1$log10FoldChange <- log10(topGOtable.select1$FoldChange)
    topGOtable.select <- topGOtable.select1[topGOtable.select1$log2FoldChange >= log2FC,]
    # Filter p.value and defined gene list
    geneList <- topGOtable.select$padj
    names(geneList) <- topGOtable.select$Genes
    geneList <- na.omit(geneList)
    #sum(topDiffGenes(geneList))
    
    #GO database from DEG selected
    geneID2GO <- as.character(factor(topGOtable.select$Gene.Ontology..GO..term))
    geneID2GO <- sub("\"", "", geneID2GO)
    geneID2GO <- sub("\\\"", "", geneID2GO)
    names(geneID2GO) <- topGOtable.select$Genes
    
    # Molecular functions
    GOdataMF <- new("topGOdata", ontology = "MF", allGenes = geneList, geneSel = topDiffGenes, 
                    nodeSize = 1, annot = annFUN.gene2GO, gene2GO = geneID2GO)
    
    # Statistics tests
    resultFisherMF <- runTest(GOdataMF, algorithm = "classic", statistic = "fisher")
    resultKSMF <- runTest(GOdataMF, algorithm = "classic", statistic = "ks")
    resultKS.elimMF <- runTest(GOdataMF, algorithm = "elim", statistic = "ks")    
    allResMF <- GenTable(GOdataMF, classicFisher = resultFisherMF, classicKS = resultKSMF, 
                         elimKS = resultKS.elimMF, topNodes = resultKS.elimMF@geneData[4])
    allResMF$GOH <- rep("MF", nrow(allResMF))
    
    # Biological Process
    GOdataBP <- new("topGOdata", ontology = "BP", allGenes = geneList, geneSel = topDiffGenes, 
                    nodeSize = 1, annot = annFUN.gene2GO, gene2GO = geneID2GO)
    # Statistics tests
    resultFisherBP <- runTest(GOdataBP, algorithm = "classic", statistic = "fisher")
    resultKSBP <- runTest(GOdataBP, algorithm = "classic", statistic = "ks")
    resultKS.elimBP <- runTest(GOdataBP, algorithm = "elim", statistic = "ks")    
    allResBP <- GenTable(GOdataBP, classicFisher = resultFisherBP, classicKS = resultKSBP, elimKS =
                           resultKS.elimBP, topNodes = resultKS.elimBP@geneData[4])
    allResBP$GOH <- rep("BP", nrow(allResBP))
    
    # cellular component
    GOdataCC <- new("topGOdata", ontology = "CC", allGenes = geneList, geneSel = topDiffGenes, 
                    nodeSize = 1, annot = annFUN.gene2GO, gene2GO = geneID2GO)
    # Statistics tests
    resultFisherCC <- runTest(GOdataCC, algorithm = "classic", statistic = "fisher")
    resultKSCC <- runTest(GOdataCC, algorithm = "classic", statistic = "ks")
    resultKS.elimCC <- runTest(GOdataCC, algorithm = "elim", statistic = "ks")    
    allResCC <- GenTable(GOdataCC, classicFisher = resultFisherCC, classicKS = resultKSCC, 
                         elimKS = resultKS.elimCC, topNodes = resultKS.elimCC@geneData[4])
    allResCC$GOH <- rep("CC", nrow(allResCC))
    
    pdf(file = paste("Graphs-GO-Functions-DEG-Islands-Inside-Up-",unique(topGOtable$Group),"-",Condition[i],"-Cladocopium-C1-Pocillopora.pdf", sep = ""), height = 18)
      showSigOfNodes(GOdataMF, score(resultKS.elimMF), firstSigNodes = 10, useInfo = 'all')
      showSigOfNodes(GOdataBP, score(resultKS.elimBP), firstSigNodes = 10, useInfo = 'all')
      showSigOfNodes(GOdataCC, score(resultKS.elimCC), firstSigNodes = 10, useInfo = 'all')
    dev.off()
    
    AllmyGO <- rbind(allResMF, allResBP, allResCC)
    AllmyGO$Group <- rep(unique(topGOtable$Group), nrow(AllmyGO))
    AllmyGO$Condition <- rep(Condition[i], nrow(AllmyGO))
    AllmyGO$Filter <- rep(paste("padj<",padj,"&log2FC>=",log2FC, sep=""), nrow(AllmyGO))
    AllmyGO$UpvsDown <- rep("Up", nrow(AllmyGO))
    write.table(AllmyGO, file = paste("GO-Functions-DEG-Islands-Inside-Up-",unique(topGOtable$Group),"-",Condition[i],"-Cladocopium-C1-Pocillopora.txt", sep = ""), quote = FALSE, sep = "\t", row.names = FALSE)
    AllmyGO.VF <- rbind(AllmyGO.VF, AllmyGO)
  }
  AllmyGO.VF
}

# Function TOPGO
topGO.fun.Down <- function(Condition, topGOtable, log2FC){
  AllmyGO.VF <- data.frame()
  for(i in 1:length(Condition)){
    topGOtable.select1 <- topGOtable[grep(Condition[i], topGOtable$Condition),]
   
    # Remove NA in pvalue
    topGOtable.select <- topGOtable.select1[!is.na(topGOtable.select1$padj),]
    topGOtable.select1$FoldChange <- 2^topGOtable.select1$log2FoldChange
    topGOtable.select1$log10FoldChange <- log10(topGOtable.select1$FoldChange)
    topGOtable.select <- topGOtable.select1[topGOtable.select1$log2FoldChange <= -log2FC,]
    # Filter p.value and defined gene list
    geneList <- topGOtable.select$padj
    names(geneList) <- topGOtable.select$Genes
    geneList <- na.omit(geneList)
    #sum(topDiffGenes(geneList))
    
    #GO database from DEG selected
    geneID2GO <- as.character(factor(topGOtable.select$Gene.Ontology..GO..term))
    geneID2GO <- sub("\"", "", geneID2GO)
    geneID2GO <- sub("\\\"", "", geneID2GO)
    names(geneID2GO) <- topGOtable.select$Genes
    
    # Molecular functions
    GOdataMF <- new("topGOdata", ontology = "MF", allGenes = geneList, geneSel = topDiffGenes, 
                    nodeSize = 1, annot = annFUN.gene2GO, gene2GO = geneID2GO)
    
    # Statistics tests
    resultFisherMF <- runTest(GOdataMF, algorithm = "classic", statistic = "fisher")
    resultKSMF <- runTest(GOdataMF, algorithm = "classic", statistic = "ks")
    resultKS.elimMF <- runTest(GOdataMF, algorithm = "elim", statistic = "ks")    
    allResMF <- GenTable(GOdataMF, classicFisher = resultFisherMF, classicKS = resultKSMF, 
                         elimKS = resultKS.elimMF, topNodes = resultKS.elimMF@geneData[4])
    allResMF$GOH <- rep("MF", nrow(allResMF))
    
    # Biological Process
    GOdataBP <- new("topGOdata", ontology = "BP", allGenes = geneList, geneSel = topDiffGenes, 
                    nodeSize = 1, annot = annFUN.gene2GO, gene2GO = geneID2GO)
    # Statistics tests
    resultFisherBP <- runTest(GOdataBP, algorithm = "classic", statistic = "fisher")
    resultKSBP <- runTest(GOdataBP, algorithm = "classic", statistic = "ks")
    resultKS.elimBP <- runTest(GOdataBP, algorithm = "elim", statistic = "ks")    
    allResBP <- GenTable(GOdataBP, classicFisher = resultFisherBP, classicKS = resultKSBP, elimKS =
                           resultKS.elimBP, topNodes = resultKS.elimBP@geneData[4])
    allResBP$GOH <- rep("BP", nrow(allResBP))
    
    # cellular component
    GOdataCC <- new("topGOdata", ontology = "CC", allGenes = geneList, geneSel = topDiffGenes, 
                    nodeSize = 1, annot = annFUN.gene2GO, gene2GO = geneID2GO)
    # Statistics tests
    resultFisherCC <- runTest(GOdataCC, algorithm = "classic", statistic = "fisher")
    resultKSCC <- runTest(GOdataCC, algorithm = "classic", statistic = "ks")
    resultKS.elimCC <- runTest(GOdataCC, algorithm = "elim", statistic = "ks")    
    allResCC <- GenTable(GOdataCC, classicFisher = resultFisherCC, classicKS = resultKSCC, 
                         elimKS = resultKS.elimCC, topNodes = resultKS.elimCC@geneData[4])
    allResCC$GOH <- rep("CC", nrow(allResCC))
    
    pdf(file = paste("Graphs-GO-Functions-DEG-Islands-Inside-Down-",unique(topGOtable$Group),"-",Condition[i],"-Cladocopium-C1-Pocillopora.pdf", sep = ""), height = 18)
      showSigOfNodes(GOdataMF, score(resultKS.elimMF), firstSigNodes = 10, useInfo = 'all')
      showSigOfNodes(GOdataBP, score(resultKS.elimBP), firstSigNodes = 10, useInfo = 'all')
      showSigOfNodes(GOdataCC, score(resultKS.elimCC), firstSigNodes = 10, useInfo = 'all')
    dev.off()
    
    AllmyGO <- rbind(allResMF, allResBP, allResCC)
    AllmyGO$Group <- rep(unique(topGOtable$Group), nrow(AllmyGO))
    AllmyGO$Condition <- rep(Condition[i], nrow(AllmyGO))
    AllmyGO$UpvsDown <- rep("Down", nrow(AllmyGO))
    AllmyGO$Filter <- rep(paste("padj<",padj,"&log2FC<=",log2FC, sep=""), nrow(AllmyGO))
    write.table(AllmyGO, file = paste("GO-Functions-DEG-Islands-Inside-Down-",unique(topGOtable$Group),"-",Condition[i],"-Cladocopium-C1-Pocillopora.txt", sep = ""), quote = FALSE, sep = "\t", row.names = FALSE)
    AllmyGO.VF <- rbind(AllmyGO.VF, AllmyGO)
  }
  AllmyGO.VF
}

# Create a Function to select DEGs from geneList
topDiffGenes <- function(x) {
  return(x < padj)
}

# Parameters to defined GO enrichment
padj <- 0.001 
```

## 2.2. G2

```{r}
topGOtableAll <- myDatalink.SA

ConditionG2.Bis <- unique(topGOtableAll$Condition)
ConditionG2 <- ConditionG2.Bis[order(ConditionG2.Bis, decreasing = FALSE)]

topGOtableVF.Up <- topGO.fun.Up(Condition = ConditionG2, topGOtable = topGOtableAll, log2FC = 0)
topGOtableVF.Down <- topGO.fun.Down(Condition = ConditionG2, topGOtable = topGOtableAll, log2FC = 0)
```


## 2.6. Combo

```{r}
AllmyGOVF.Up.Down <- rbind(topGOtableVF.Up, topGOtableVF.Down)
AllmyGOVF.Up.Down$Diff.Significants.Expected <- AllmyGOVF.Up.Down$Significant - AllmyGOVF.Up.Down$Expected

write.table(AllmyGOVF.Up.Down, file ="Functions-GO-DEG-C1-Analyses-by-Groups-Up-Down.txt", quote = FALSE, sep = "\t", row.names = FALSE)

AllmyGOVF <- read.table("Functions-GO-DEG-C1-Analyses-by-Groups-Up-Down.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE, quote = "")
AllmyGOVF$Conditions <- paste(AllmyGOVF$GOH, AllmyGOVF$Group, AllmyGOVF$Condition, AllmyGOVF$UpvsDown, sep = "-")
AllmyGOVF$Diff.Significants.Expected2 <- abs(AllmyGOVF$Diff.Significants.Expected)
AllmyGOVF$Colors <- rep("Colors", nrow(AllmyGOVF))
AllmyGOVF$Colors[AllmyGOVF$Diff.Significants.Expected >= 0] <- "Positive"
AllmyGOVF$Colors[AllmyGOVF$Diff.Significants.Expected <= 0] <- "Negative"

pdf(file = "DEG-GO-C1-BarPlot-TOP25-by-C1-GROUPS.pdf", width = 24, height = 14)
for(i in 1:length(unique(AllmyGOVF$Conditions))){
  enrGo.DEG.PFAM.Select1 <- AllmyGOVF[grep(unique(AllmyGOVF$Conditions)[i], AllmyGOVF$Conditions),]
  enrGo.DEG.PFAM.Select <- enrGo.DEG.PFAM.Select1[enrGo.DEG.PFAM.Select1$Diff.Significants.Expected2 != 0,]
  enrGo.DEG.Order.PFAM.Select <- enrGo.DEG.PFAM.Select[order(enrGo.DEG.PFAM.Select$Diff.Significants.Expected2, decreasing = TRUE),]
  if(nrow(enrGo.DEG.Order.PFAM.Select) != 0){
    plot(ggplot(enrGo.DEG.Order.PFAM.Select[1:25,], aes(x = Diff.Significants.Expected, y = Term, fill = Colors ))+geom_bar(stat="identity", color="grey", fill="white")+labs(title = paste("Top 25 GO terms in Cladocopium C1 in", unique(enrGo.DEG.Order.PFAM.Select$Conditions), sep =" "), y = "PFAM", x = "Number of genes enriched")+theme(strip.text.y = element_text(size = 15), axis.text.y = element_text(size = 13), axis.text.x = element_text(size = 13)))
    }
}
dev.off()
```
